astropy/time/tests/test_custom_formats.py

```diff
@@ -0,0 +1,26 @@
+import pytest
+
+from astropy.time import Time, TimeFormat
+
+
+def test_custom_time_format_exception():
+    class SpecificException(ValueError):
```

Comment:
I'd take this outside - fine to define it locally in the test module.

Reply:
My thought was that I wanted this exception to be, well, specific to this function - no other code anywhere ought to be able to make an exception that would satisfy this test. There's no need to be quite so strict, though, and I suppose one ought to test generation of SpecificExceptions in all the various places CustomFormat might want code. (So far I think in val/val2 type validation, in construction, and in value extraction? The last poses no particular difficulty.)